lib/Worker.js

```diff
@@ -68,7 +68,7 @@ module.exports = (function(){
                 return sessionManager.refreshSessionTimestamp(e.entryId);
             })
             .then(function() {
-                var entryDownloader = new entryDownloaderCtor(e.entryId, that.hostname, that.port, that.applicationName, e.dvrWindow, e.maxChunkCount);
+                var entryDownloader = new entryDownloaderCtor(e.entryId, that.hostname, that.port, that.applicationName, e.flavorParamsIds.split(","), e.dvrWindow, e.maxChunkCount, true);
```

Comment:
Please pass e object as a parameter to the entry download instead of 100 different params...